migrations/20220506135000_schema_breakdown_ddl.sql

```diff
@@ -0,0 +1,72 @@
+CREATE TABLE IF NOT EXISTS `type_def_types` (
+    `id` int UNSIGNED NOT NULL AUTO_INCREMENT,
+    `name` VARCHAR(255) NOT NULL,
+    `description` TEXT,
+    `type` ENUM('OBJECT', 'SCALAR', 'INTERFACE', 'ENUM', 'INPUT', 'UNION', 'DIRECTIVE') NOT NULL,
+    PRIMARY KEY (`id`),
+    UNIQUE KEY `name` (`name`)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_general_ci;
+
+CREATE TABLE IF NOT EXISTS `type_def_fields` (
+    `id` INT UNSIGNED NOT NULL AUTO_INCREMENT,
+  	`name` VARCHAR(255) NOT NULL,
+    `description` TEXT,
+    `is_nullable` tinyint(1) NOT NULL DEFAULT 1,
+    `is_array` tinyint(1) NOT NULL DEFAULT 0,
+    `is_array_nullable` tinyint(1) NOT NULL DEFAULT 1,
+    `is_deprecated` tinyint(1) NOT NULL DEFAULT 0,
+  	`parent_type_id` INT UNSIGNED,
+	`children_type_id` INT UNSIGNED,
+    PRIMARY KEY (`id`),
+  	FOREIGN KEY (`parent_type_id`) REFERENCES `type_def_types`(`id`) ON DELETE CASCADE,
+    FOREIGN KEY (`children_type_id`) REFERENCES `type_def_types`(`id`) ON DELETE CASCADE,
+    UNIQUE KEY `name` (`name`, `parent_type_id`)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_general_ci;
+
+CREATE TABLE IF NOT EXISTS `type_def_operations` (
+    `id` INT UNSIGNED NOT NULL AUTO_INCREMENT,
+    `name` VARCHAR(255) NOT NULL,
+    `description` TEXT,
+    `type` ENUM('QUERY', 'MUTATION', 'SUBSCRIPTION') NOT NULL,
+    `service_id` INT UNSIGNED NOT NULL,
+    PRIMARY KEY (`id`),
+    FOREIGN KEY (`service_id`) REFERENCES `services`(`id`) ON DELETE CASCADE,
+    UNIQUE KEY `name` (`name`, `type`)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_general_ci;
+
+CREATE TABLE IF NOT EXISTS `type_def_operation_parameters` (
+    `operation_id` INT UNSIGNED,
+    `type_id` INT UNSIGNED,
+    `name` VARCHAR(255) NOT NULL,
+    `description` TEXT,
```

Comment:
maybe limit this to small text or medium text?

Reply:
same in other types

Reply:
We agree on this matter and we've changed it from TEXT to TINYTEXT.